docs/health.md

```diff
@@ -18,3 +18,70 @@ with at least one value for `hostname` or `IP`.
 
 ### PersistentVolumeClaim
 * The `status.phase` is `Bound`
+
+## Health Checks Customization
+
+Argo CD leverages [Lua](https://www.lua.org/) script to implement health assessment for several standard Kubernetes types. This allows to quickly customized health assessments
+and provide health checks for Kubernetes custom resources.
+
+There are two ways to include custom health check.
+
+### Define Custom Health Check in `argocd-cm` ConfigMap
```

Comment:
Should we have a centralized configuration guide for `argocd-cm`? It's all spread out at the moment.